model/lifecycle.go

```diff
@@ -969,6 +969,14 @@ func makeDeps(t BuildVariantTaskUnit, thisTask *task.Task, taskIds TaskIdTable)
 			if id == thisTask.Id {
 				continue
 			}
+			if id == "" {
```

Comment:
I'm actually thinking that this can't be the place where blank ids show up, since all of the above functions check "". Is there anywhere else we add dependencies? (In the DB, do we have tasks that actually have blank IDs? or else i wonder if there's a logic bug that makes it seem like there are tasks that do)

Reply:
that's a good point: the example task we are looking at doesn't have any empty taskIds in the DB 
![image](https://user-images.githubusercontent.com/13104717/173376943-975ddafc-7b90-4f57-b779-57148ee915a6.png)